src/main/java/at/ac/tuwien/damap/conversion/AbstractTemplateExportScienceEuropeComponents.java

```diff
@@ -53,244 +53,272 @@ public void determinteDatasetIDs(){
-            }
-            //variable project funding, combination from funding item variables
-            if (!fundingItems.isEmpty()) {
-                addReplacement(replacements, "[grantid]", String.join(", ", fundingItems));
-            }
-            else {
-                addReplacement(replacements, "[grantid]", "");
-            }
+        // add funding program to funding item variables
+        titlePageFunding(project, projectCRIS);
 
-            //variable project ID
-            addReplacement(replacements, "[projectid]", dmp.getProject().getUniversityId());
+        // variable project ID
+        addReplacement(replacements, "[projectid]", project.getUniversityId());
+    }
+
+    private void titlePageFunding(Project project, ProjectDO projectCRIS) {
+        List<String> fundingItems = new ArrayList<>();
+        if (projectCRIS != null && projectCRIS.getFunding() != null &&
+                projectCRIS.getFunding().getFundingProgram() != null) {
+            fundingItems.add(projectCRIS.getFunding().getFundingProgram());
+
+        }
+        // add grant number to funding item variables
+        if (project.getFunding() != null && project.getFunding().getGrantIdentifier().getIdentifier() != null) {
+            fundingItems.add(project.getFunding().getGrantIdentifier().getIdentifier());
+        }
+        // variable project funding, combination from funding item variables
+        if (!fundingItems.isEmpty()) {
+            addReplacement(replacements, "[grantid]", String.join(", ", fundingItems));
+        } else {
+            addReplacement(replacements, "[grantid]", "");
         }
     }
 
-    public void contributorInformation(){
-        //mapping contact information
-        if (dmp.getContact() != null) {
-            List<String> contactItems = new ArrayList<>();
-            String contactName = "";
-            String contactMail = "";
-            String contactId = "";
-            String contactIdentifierType = "";
-            String contactIdentifierId = "";
-            String contactAffiliation = "";
-            String contactAffiliationId = "";
-            String contactAffiliationIdentifierType = "";
-            String contactAffiliationIdentifierId = "";
-
-            if (dmp.getContact().getFirstName() != null && dmp.getContact().getLastName() != null) {
-                contactName = dmp.getContact().getFirstName() + " " + dmp.getContact().getLastName();
-                contactItems.add(contactName);
-            }
+    private String getContributorPersonIdentifier(Contributor contributor) {
+        String identifier = null;
+        Identifier personIdentifier = contributor.getPersonIdentifier();
 
-            if (dmp.getContact().getMbox() != null) {
-                contactMail = dmp.getContact().getMbox();
-                contactItems.add(contactMail);
+        if (personIdentifier != null) 
+        {
+            String contactIdentifierId = personIdentifier.getIdentifier();
```

Comment:
Unnecessary local variable

Reply:
same reason as with affiliation, to keep it open for future identifiers